### PR TITLE
Video upload

### PR DESCRIPTION
**Pull request by CorneliaKelinske**
Hello @alanvardy , I think it's time to review what has been done so far and merge this branch in, if no more changes are necessary for now.

I am now showing different error messages on my file upload form depending on whether
a) no file has been chosen
b) description and/or title are missing
c) invalid file format has been chosen
d) file path is invalid

Quick review please! tests are still all passing :)

### CODE REVIEW

File: lib/the_little_thinkers_space/content/upload.ex [hunk 1]
```diff
@@ -1,7 +1,22 @@
...
   use Ecto.Schema
   import Ecto.Changeset
 
+  @valid_image_types ["image/jpeg", "image/jpg", "image/png"]
+  @valid_video_types ["video/mp4", "video/quicktime"]
+  @valid_file_types @valid_image_types ++ @valid_video_types
+
+  def valid_image_types, do: @valid_image_types
+  def valid_video_types, do: @valid_video_types
+
```
Comment (alanvardy) [inline lib/the_little_thinkers_space/content/upload.ex:17]: :scissors:

File: lib/the_little_thinkers_space_web/controllers/upload_controller.ex [hunk 2]
```diff
@@ -16,12 +16,30 @@ defmodule TheLittleThinkersSpaceWeb.UploadController do
...
+        "title" => title,
+        "description" => description,
+      }
+    },
+    _current_user
+  ) when title == "" or description == "" do
+    conn
+    |> put_flash(:error, "Please enter a file name and/or description.")
+    |> redirect(to: Routes.upload_path(conn, :new))
+  end
```
Comment (alanvardy) [inline lib/the_little_thinkers_space_web/controllers/upload_controller.ex:32]: While this works, I think that these errors would be better handled as part of the changeset.

File: lib/the_little_thinkers_space_web/controllers/upload_controller.ex [hunk 3]
```diff
@@ -39,26 +57,37 @@ defmodule TheLittleThinkersSpaceWeb.UploadController do
     else
       {:error, %Ecto.Changeset{} = changeset} ->
         # TODO show errors on page
-        render(conn, "new.html", changeset: changeset)
+        # IO.inspect(changeset)
```
Comment (alanvardy) [inline lib/the_little_thinkers_space_web/controllers/upload_controller.ex:60]: :scissors:

File: lib/the_little_thinkers_space_web/controllers/upload_controller.ex [hunk 4]
```diff
@@ -39,26 +57,37 @@ defmodule TheLittleThinkersSpaceWeb.UploadController do
...
 
-      {:error, error} ->
-        # TODO make this an error that can be rendered in new page
-        raise "Could not read file #{path <> filename}"
+        conn
+        |> put_flash(:error, "Invalid file format")
+        |> render("new.html", changeset: changeset)
+
+      #
```
Comment (alanvardy) [inline lib/the_little_thinkers_space_web/controllers/upload_controller.ex:66]: :scissors:

File: lib/the_little_thinkers_space_web/controllers/upload_controller.ex [hunk 5]
```diff
@@ -4,7 +4,7 @@ defmodule TheLittleThinkersSpaceWeb.UploadController do
   alias TheLittleThinkersSpace.Content
   alias TheLittleThinkersSpace.Content.Upload
 
-  plug :authenticate_user when action in [:index, :show, :new]
+  plug :authenticate_user when action in [:index, :show, :new, :create, :edit]
```
Comment (alanvardy) [inline lib/the_little_thinkers_space_web/controllers/upload_controller.ex:7]: I think that you need to add `:update` to this list as well.

File: lib/the_little_thinkers_space_web/templates/upload/form.html.eex [hunk 6]
```diff
@@ -13,11 +9,11 @@
 
   <%= label f, :title %>
   <%= text_input f, :title %>
-  <%= error_tag f, :title %>
+
 
   <%= label f, :description %>
   <%= textarea f, :description %>
-  <%= error_tag f, :description %>
+
```
Comment (alanvardy) [inline lib/the_little_thinkers_space_web/templates/upload/form.html.eex:16]: You may wish to keep these error tags if you wish to display changeset errors

File: lib/the_little_thinkers_space/content/upload.ex (hunk 1 above)
Comment (CorneliaKelinske) [reply lib/the_little_thinkers_space/content/upload.ex:17]: done

File: lib/the_little_thinkers_space/content/upload.ex [hunk 7]
```diff
@@ -30,7 +30,7 @@ defmodule TheLittleThinkersSpace.Content.Upload do
   def changeset(upload, attrs) do
     upload
     |> cast(attrs, [:file, :file_type, :title, :description])
-    |> validate_required([:file, :file_type, :title, :description])
-    |> validate_inclusion(:file_type, @valid_file_types)
+    |> validate_required([:file, :file_type, :title, :description], message: "This box must not be emtpy!")
```
Comment (alanvardy) [inline lib/the_little_thinkers_space/content/upload.ex:33]: Spelling (empty)

Comment (CorneliaKelinske) [reply lib/the_little_thinkers_space/content/upload.ex:33]: Thank you!

**Pull request merged**